ansible/playbooks/roles/prometheus/tasks/upgrade/main.yml

```
@@ -0,0 +1,122 @@
+---
+# General things needed 
+- name: Prometheus | Include new version default variables
+  include_vars:
+    file: defaults/main.yml
+    name: prometheus_defaults
+
+- name: Prometheus | Populate service facts
+  service_facts:
+  when: ansible_facts.service is undefined
+  no_log: true
+
+
+# Prometheus upgrade
+- name: Prometheus upgrade | Set prometheus_will_upgrade fact
+  set_fact:
+    prometheus_will_upgrade: false
+
+- name: Prometheus upgrade | Check binary existance
+  stat:
+    path: /usr/local/bin/prometheus
+  register: prometheus_stat
+
+- name: Prometheus upgrade | Start upgrade
+  when:
+    - ansible_facts.services["prometheus.service"] is defined
+    - prometheus_stat.stat.exists
+  block:
+    - name: Prometheus upgrade | Get version
+      command: prometheus --version
+      changed_when: false
+      check_mode: false
+      register: __prometheus_version
+
+    # The following issue is also present in prometheus itself and fixed in newer versions:
+    # https://github.com/prometheus/node_exporter/issues/1271
+    # So we check both stderr and stdout just in case.
+    - name: Prometheus upgrade | Set version fact (stderr)
+      set_fact:
+        prometheus_version: "{{ __prometheus_version.stderr_lines[0].split(' ')[2] }}"
+      when: __prometheus_version.stderr_lines is defined and (__prometheus_version.stderr_lines|length>0)
```

```suggestion
    - name: Set version fact
      set_fact:
        prometheus_version: "{{ _output_lines[0].split(' ')[2] }}"
      vars:
        _output_lines: "{{ _prometheus_version.stdout_lines + _prometheus_version.stderr_lines }}"
```
How about using one task instead of two (not tested)?